packages/data-transport-layer/src/services/server/service.ts

```diff
@@ -156,8 +156,7 @@ export class L1TransportServer extends BaseService<L1TransportServerOptions> {
    * TODO: Link to our API spec.
    */
   private _registerAllRoutes(): void {
-    // TODO: Maybe add doc-like comments to each of these routes?
-
+    // TODO: this needs a backend argument
     this._registerRoute(
       'get',
```

Comment:
@smartcontracts Any thoughts on how to reduce complexity in this endpoint? We need to add the `backend` query param to this so that it can determine whether or not its sync'd L1 or L2. Ideally we refactor the db handlers now that we are moving away from verified/not verified as concepts in this codebase

Reply:
is `backend` in this case `defaultSource` or a different concept?

Reply:
> what do you think is too complex here? This looks OK to me.

I'm talking about the crazy if/else nesting
https://github.com/ethereum-optimism/optimism/pull/479/files#diff-29a43f33722b902c31ee7224a372262d2b73f170ed0af5be6aca1b54aa505a72L183-L211